pulsar-broker/src/test/java/org/apache/pulsar/client/api/DeadLetterTopicTest.java

```diff
@@ -256,4 +257,29 @@ public void testDeadLetterTopicByCustomTopicName() throws Exception {
         assertNull(checkMessage);
         checkConsumer.close();
     }
+
+    @Test(timeOut = 200000)
+    public void testDeadLetterWithoutConsumerReceiveImmediately() throws PulsarClientException, InterruptedException {
+        final String topic = "persistent://my-property/my-ns/dead-letter-topic-without-consumer-receive-immediately";
+
+        Consumer<byte[]> consumer = pulsarClient.newConsumer()
+                .topic(topic)
+                .subscriptionType(SubscriptionType.Shared)
+                .subscriptionName("my-subscription")
+                .ackTimeout(3, TimeUnit.SECONDS)
+                .deadLetterPolicy(DeadLetterPolicy.builder().maxRedeliverCount(2).build())
+                .subscribe();
+
+        Producer<byte[]> producer = pulsarClient.newProducer()
+                .topic(topic)
+                .create();
+
+        producer.send(("a message").getBytes());
+
+        // wait a while, message will send to dead letter topic
+        Thread.sleep(15000L);
```

Comment:
can we reduce the ackTimeout to 1 second or even lower, so that we can reduce the time here?